### PR TITLE
Fix UUID/ProviderID to only use lower case based on VC and k8s matching/lookup

### PR DESCRIPTION
**Pull request by dvonthenen**
**What this PR does / why we need it**:
This fixes an issue where a user-defined vSphere ProviderID `--provider-id=` is of mixed case (lower/upper). The internal workings of the cloud provider interface discovers the ProviderID in lower case and the internal lookup fails if with at least 1 UpperCase character when user-provider.

Additional changes:
- Added negative go tests for cloud provider Instance interface
- Purposely made provided UUID ToUpper in certain go tests to validate this fix
- Fixed Infof -> Info in certain locations

**Which issue this PR fixes**: https://github.com/kubernetes/cloud-provider-vsphere/issues/204

**Special notes for your reviewer**:
Tested on vSphere 6.7u2

**Release note**:
Does not affect user documentation or behavior

### CODE REVIEW

File: pkg/cloudprovider/vsphere/instances_test.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ import (
 	"testing"
 
 	"github.com/vmware/govmomi/simulator"
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (yastij) [inline pkg/cloudprovider/vsphere/instances_test.go]: nit: is this change needed ?

File: pkg/cloudprovider/vsphere/util.go [hunk 2]
```diff
@@ -34,7 +34,7 @@ const (
 
 // GetUUIDFromProviderID returns a UUID from the supplied cloud provider ID.
 func GetUUIDFromProviderID(providerID string) string {
-	return strings.TrimPrefix(providerID, ProviderPrefix)
+	return strings.ToLower(strings.TrimSpace(strings.TrimPrefix(providerID, ProviderPrefix)))
```
Comment (yastij) [inline pkg/cloudprovider/vsphere/util.go]: nit: this is becoming a bit long

File: pkg/common/connectionmanager/search.go [hunk 3]
```diff
@@ -202,12 +202,14 @@ func (cm *ConnectionManager) WhichVCandDCByNodeID(ctx context.Context, nodeID st
 					hostName = myNodeID
 				}
 
+				UUID := strings.ToLower(strings.TrimSpace(oVM.Summary.Config.Uuid))
```
Comment (yastij) [inline pkg/common/connectionmanager/search.go]: nit: lowercase `UUID`

File: pkg/common/connectionmanager/search.go [hunk 4]
```diff
@@ -202,12 +202,14 @@ func (cm *ConnectionManager) WhichVCandDCByNodeID(ctx context.Context, nodeID st
...
+
 				klog.V(2).Infof("Found node %s as vm=%+v in vc=%s and datacenter=%s",
 					nodeID, vm, res.vc, res.datacenter.Name())
-				klog.V(2).Info("Hostname: ", hostName, " UUID: ", oVM.Summary.Config.Uuid)
+				klog.V(2).Info("Hostname: ", hostName, " UUID: ", UUID)
```
Comment (yastij) [inline pkg/common/connectionmanager/search.go]: same

File: pkg/common/connectionmanager/search.go [hunk 5]
```diff
@@ -202,12 +202,14 @@ func (cm *ConnectionManager) WhichVCandDCByNodeID(ctx context.Context, nodeID st
...
 
 				vmInfo = &VMDiscoveryInfo{DataCenter: res.datacenter, VM: vm, VcServer: res.vc,
-					UUID: oVM.Summary.Config.Uuid, NodeName: hostName}
+					UUID: UUID, NodeName: hostName}
```
Comment (yastij) [inline pkg/common/connectionmanager/search.go]: same

File: pkg/common/connectionmanager/search.go (hunk 3 above)
Comment (dvonthenen) [reply pkg/common/connectionmanager/search.go]: go lint complains if UUID isn't in caps

File: pkg/common/connectionmanager/search.go (hunk 5 above)
Comment (dvonthenen) [reply pkg/common/connectionmanager/search.go]: Same comment as above

File: pkg/common/connectionmanager/search.go (hunk 4 above)
Comment (dvonthenen) [reply pkg/common/connectionmanager/search.go]: Same as above

File: pkg/cloudprovider/vsphere/instances_test.go (hunk 1 above)
Comment (dvonthenen) [reply pkg/cloudprovider/vsphere/instances_test.go]: The go lint seems to have added that in to remove ambiguity. Each time I try to save the file without it, it adds it back in.

File: pkg/cloudprovider/vsphere/util.go (hunk 2 above)
Comment (dvonthenen) [reply pkg/cloudprovider/vsphere/util.go]: Broke up the nesting of calls a little.

**Comment by dvonthenen** [comment]
Updated based on feedback from @yastij

**Comment by dvonthenen** [comment]
CC: @fabio @dougm for another look

File: pkg/common/connectionmanager/search.go (hunk 4 above)
Comment (frapposelli) [reply pkg/common/connectionmanager/search.go]: Just a nit, I like using printf style formatting instead of appending variables, it would make it consistent with the rest of the code.

Comment (dvonthenen) [reply pkg/common/connectionmanager/search.go]: Yup. Updated for uniformity.

**Comment by dvonthenen** [comment]
Updated based on @frapposelli feedback
